### PR TITLE
feat: block pool mining if not synced

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
block pool mining if we are more than 10 blocks out with a sync, 
This will allow the node to solo mine, but it wont mine p2pool blocks

### CODE REVIEW

File: src/server/grpc/p2pool.rs [hunk 1]
```diff
@@ -121,17 +123,28 @@ where S: ShareChain
...
     /// Submits a new block to share chain and broadcasts to the p2p network.
     pub async fn submit_share_chain_block(&self, block: Arc<P2Block>) -> Result<(), Status> {
-        if !self.are_we_synced_with_p2pool.load(Ordering::Relaxed) {
-            info!(target: LOG_TARGET, "We are not synced yet, not submitting block atm");
-            return Ok(());
-        }
         let pow_algo = block.original_header.pow.pow_algo;
+        match pow_algo {
+            PowAlgorithm::RandomX => {
+                if !self.are_we_synced_with_randomx_p2pool.load(Ordering::Relaxed) {
```
Comment (stringhandler) [inline src/server/grpc/p2pool.rs]: Use the most strict one, I think it's SeqCt or something

```suggestion
                if !self.are_we_synced_with_randomx_p2pool.load(Ordering::Seq) {
```

File: src/server/p2p/network.rs [hunk 2]
```diff
@@ -1753,6 +1761,17 @@ where S: ShareChain
...
         }
 
+        // get the sync bool
+        let sync_status = match algo {
+            PowAlgorithm::RandomX => self.are_we_synced_with_randomx_p2pool.clone(),
+            PowAlgorithm::Sha3x => self.are_we_synced_with_sha3x_p2pool.clone(),
+        };
+        let our_tip = share_chain.get_tip().await?.unwrap_or_default();
+        if our_tip.0 < their_height.saturating_sub(10) {
+            info!(target: SYNC_REQUEST_LOG_TARGET, "We({}) are out by more than 10 blocks from syncing peer({}), seeting sync status to false", our_tip.0, their_height);
```
Comment (stringhandler) [inline src/server/p2p/network.rs]:
```suggestion
            info!(target: SYNC_REQUEST_LOG_TARGET, "We({}) are out by more than 10 blocks from syncing peer({}), setting sync status to false", our_tip.0, their_height);
```

**Review by stringhandler — COMMENTED** [review]
The most important part of this is paying the correct shares, but that seems to be missing

**Review by stringhandler — COMMENTED** [review]
I also can't see where you set it to true, other than after network silence

**Comment by SWvheerden** [comment]
> I also can't see where you set it to true, other than after network silence

Diff is not showing it, but its set after catchup sync completes. You only see the change in name, its called `synced_bool`

**Review by stringhandler — APPROVED** [review]

**Pull request merged**